middleware/proxy/proxy.go

```diff
@@ -123,11 +123,13 @@ func Do(c *fiber.Ctx, addr string) error {
 	res := c.Response()
 	originalURL := utils.CopyString(c.OriginalURL())
 	defer req.SetRequestURI(originalURL)
-	req.SetRequestURI(addr)
+
+	copiedURL := utils.CopyString(addr)
+	req.SetRequestURI(copiedURL)
```

Comment:
```suggestion
	req.SetRequestURI(utils.CopyString(addr))
```
this should be enough, because within the string is changed

Reply:
i was wrong -> your code is okay